archive/demos/shiny-drum-machine-ui.js

```diff
@@ -0,0 +1,272 @@
+        this.step = step;
+    }
+
+    set value(value) {
+        this.labelElement.innerText = Math.min(this.max, Math.max(this.min, value));
+    }
+
+    get value() {
+        return new Number(this.labelElement.innerText);
+    }
+}
+
+class Notes {
+    constructor() {
+        this.onClick = (instrument, rhythm) => {};
+
+        this.buttons = {};
+        for (const el of document.querySelectorAll(`[data-instrument][data-rhythm]`)) {
+            const instrument = el.dataset.instrument;
+            const rhythm = new Number(el.dataset.rhythm);
+
+            if(!this.buttons[instrument]) {
+                this.buttons[instrument] = {};
+            }
+
+            this.buttons[instrument][rhythm] = new Button(el);
+            this.buttons[instrument][rhythm].onclick = () => this.onClick(instrument, rhythm);
+        }
+    }
+
+    setNote(instrument, rhythmIndex, note) {
+        this.buttons[instrument][rhythmIndex].state = note;
+    }
+}
+
+class Modal {
+    constructor(element) {
+        this.element = element;
+    }
+
+    toggleVisibility() {
+        document.getElementById('pad').classList.toggle('active');
+        document.getElementById('params').classList.toggle('active');
+        document.getElementById('tools').classList.toggle('active');
+        this.element.classList.toggle('active');
+    }
+}
+
+class SaveModal extends Modal {
+    constructor(getDataCallback) {
+        super(document.getElementById('save_container'));
+        this.textarea = document.getElementById('save_textarea');
+        document.getElementById('save_ok').addEventListener('click', () => {
+            this.toggleVisibility();
+        });
+        document.getElementById('save').addEventListener('click', () => {
+            this.textarea.value = getDataCallback();
+            this.toggleVisibility();
+        });
+    }
+
+    show(data) {
+        this.textarea.value = data;
+        this.toggleVisibility();
+    }
+}
+
+class LoadModal extends Modal {
+    constructor() {
+        super(document.getElementById('load_container'));
+        this.textarea = document.getElementById('load_textarea');
+        this.onLoad = (data) => {};
+        document.getElementById('load_ok').addEventListener('click', () => {
+            this.onLoad(this.textarea.value);
+            this.toggleVisibility();
+            this.textarea.value = "";
+        });
+        document.getElementById('load_cancel').addEventListener('click', () => {
+            this.toggleVisibility();
+        });
+        document.getElementById('load').addEventListener('click', () => {
+            this.toggleVisibility();
+        });
+    }
+
+    show(data) {
+        this.textarea.value = data;
+        this.toggleVisibility();
+    }
+}
+
+export {DemoButtons,
+     EffectPicker,
+     KitPicker,
+     EffectSlider,
+     SwingSlider,
+     PitchSliders,
+    TempoInput, Playheads, Notes, PlayButton,
+    LoadModal, SaveModal, ResetButton };
```

Comment:
Can we add one newline at the end? Also the indentation looks a bit weird. Perhaps eslint-config-google will help?